superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -489,10 +490,15 @@ export default class ResultSet extends React.PureComponent<
       return <Alert bsStyle="warning">Query was stopped</Alert>;
     }
     if (query.state === 'failed') {
+      const title =
+        query?.errors?.[0].error_type ===
+        ErrorTypeEnum.MISSING_TEMPLATE_PARAMS_ERROR
+          ? t('Parameter Error')
+          : t('Database Error');
```

Comment:
yeah, ideally this logic would be handled by the error component registry, but since we're not using the new error format for all SQL Lab errors, that's not possible yet. Maybe add a todo comment to clean this up?

Reply:
Yeah, I'll leave a TODO here.